docs/user-guides/retrieval.md

```diff
@@ -0,0 +1,282 @@
+# Retrieval in CLIP-as-service
+`CLIP-as-service` offers us high-quality embeddings. Retrieval is one of the most common use cases for embeddings. Retrieval in `CLIP-as-service` support indexing a very large dataset (millions/billions) and querying within 50ms, depending on the machine.
+
+In order to implement retrieval, we add an [`AnnLite`](https://github.com/jina-ai/annlite) indexer executor after the encoder executor in CLIP-as-service.
+
+
+## Fast search in CLIP-as-service
+Indexing and searching are easy in `CLIP-as-service`:
+
+```python
+from clip_client import Client
+
+client = Client('grpc://0.0.0.0:23456')
+
+# index
+client.index(
+    [
+        Document(text='she smiled, with pain'),  # text
+        Document(uri='apple.png'),  # local image
+        Document(
+            uri='https://clip-as-service.jina.ai/_static/favicon.png'
+        ),  # online image
+    ]
+)
+
+# search
+client.search(['smile'])
+```
+
+The results will look like this:
```

Comment:
What does this image/section tell me as a user? -> Not quite clear to me, but i don't have a lot of experience with Jina yet